src/lib/operators/join_sort_merge.cpp

```diff
@@ -87,7 +91,8 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
         _right_column_id{right_column_id},
         _op{op},
         _mode{mode} {
-    _cluster_count = _determine_number_of_clusters();
+    _cluster_count = _determine_number_of_clusters(_sort_merge_join.input_table_left()->row_count(),
+                                                   _sort_merge_join.input_table_right()->row_count());
```

Comment:
This is weirdly coupled with the non-Impl class? Any chance that should happen there and be passed into the Impl?